File: Leetcode/150-Evaluate-Reverse-Polish-Notation/150-Evaluate-Reverse-Polish-Notation.py
class Solution:
    def evalRPN(self, tokens):
        stack = []

        for op in tokens:
            if op in {"+", "-", "*", "/"}:
                b = stack.pop()
                a = stack.pop()

                if op == "+":
                    stack.append(a + b)
                elif op == "-":
                    stack.append(a - b)
                elif op == "*":
                    stack.append(a * b)
                elif op == "/":
                    stack.append(int(a / b))
            else:
                    stack.append(int(op))

        return stack[0]

File: Leetcode/150-Evaluate-Reverse-Polish-Notation/150-Evaluate-Reverse-Polish-Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int operand1, operand2;

        for (String token : tokens) {
            switch (token) {
                case "+":
                    operand2 = stack.pop();
                    operand1 = stack.pop();
                    stack.push(operand1 + operand2);
                    break;
                case "*":
                    operand2 = stack.pop();
                    operand1 = stack.pop();
                    stack.push(operand1 * operand2);
                    break;
                case "/":
                    operand2 = stack.pop();
                    operand1 = stack.pop();
                    stack.push(operand1 / operand2);
                    break;
                case "-":
                    operand2 = stack.pop();
                    operand1 = stack.pop();
                    stack.push(operand1 - operand2);
                    break;
                default:
                    stack.push(Integer.parseInt(token));
            }
        }

        return stack.pop();
    }
}